src/tribler-core/tribler_core/session.py

```diff
@@ -73,7 +70,9 @@ class Session(TaskManager):
     """
     __single = None
 
-    def __init__(self, config, core_test_mode=False):
+    def __init__(self, config: TriblerConfig, core_test_mode: bool = False,
```

Comment:
There are now config arguments next to a config object. These should not be apart.

Reply:
Agree, but `core_test_mode` comes from: https://github.com/drew2a/tribler/blob/ada438d691d1e14b3da743f0c2f6e7a6cc8fa7fb/src/run_tribler.py#L178

So, it is not a part of Tribler's config.

Reply:
`core_test_mode`, `chant_testnet`, `trustchain_testnet` and `community_loader` all serve to configure a `Session`. Therefore, they should be part of the configuration.

I know `core_test_mode` was already there, even though it shouldn't be. However, that doesn't justify adding more of a bad thing.

Reply:
All variables that are passed into a class constructor are some kind of configuration for this particular class.

There is no reason to always use only one variable for this purpose.

If there are too many variables, of course then you can consolidate them.

Reply:
There are always exceptions to a rule. However, in this case, you're passing additional boolean values to enable/disable functionality next to an object (`TriblerConfig`) that exists to capture the configuration of components, including enabling/disabling functionality.

`chant_testnet` and `trustchain_testnet` specify ways of loading respectively the channel and trustchain components, they belong next to the other settings to configure these components. These are not special configuration options, they require no obscure datatypes or initialization. These are just boolean configuration values.

Reply:
Unfortunately `chant_testnet` and `trustchain_testnet` are not a part of Tribler config, but calculated as:

```python
('TESTNET' in os.environ or
            'CHANT_TESTNET' in os.environ or
            config.chant.testnet)
```

and 

```python
('TESTNET' in os.environ or
            'TRUSTCHAIN_TESTNET' in os.environ or
            config.trustchain.testnet)
```

So, they are special.

In that case, how would you like them to be transferred to the session?

Reply:
Configuration should be set in the configuration object. A Session should not have to worry about fixing its configuration.

I consider this the perfect solution:

https://github.com/Tribler/tribler/blob/43a428d3b0bee3256c559d5cb0cbb1d4c6091c12/src/tribler-core/tribler_core/modules/metadata_store/config.py#L10

🔽 

```python
   testnet: bool = 'TESTNET' in os.environ or 'CHANT_TESTNET' in os.environ
```

Reply:
As far as I know, by doing `testnet: bool = 'TESTNET' in os.environ or 'CHANT_TESTNET' in os.environ` in  `MetadataStoreConfig` definition you specify a default value of `testnet` variable.

But `'TESTNET' in os.environ or 'CHANT_TESTNET' in os.environ` should be calculated in a runtime and be a disjunction with testnet value.

But wait.. probably I know the way. Wait a minute...

Reply:
I can do something like:

```python
class BandwidthAccountingSettings(TriblerConfigSection):
    testnet: bool = Field(..., env='CHANT_TESTNET')
```

https://pydantic-docs.helpmanual.io/usage/settings/#environment-variable-names

Reply:
Perfect 👍

Reply:
@qstokkink done. Please, check it: 

https://github.com/Tribler/tribler/pull/6182/commits/5e99b28359e4d7721169fa2268b7e5dd1c87d672